sdk/resourcemanager/Azure.ResourceManager.Core/tests/Unit/ArmClientOptionsTests.cs

```diff
@@ -33,5 +33,44 @@ public void TestClientOptionsParamCheck()
             var options = new AzureResourceManagerClientOptions();
             Assert.Throws<ArgumentNullException>(() => { options.AddPolicy(null, HttpPipelinePosition.PerCall); });
         }
+
+        [TestCase]
+        public void VersionLoadedChanges()
+        {
+            AzureResourceManagerClientOptions options = new AzureResourceManagerClientOptions();
+
+            options.FakeRestApiVersions().FakeResourceVersion = FakeResourceApiVersions.V2019_12_01;
+            var result = options.ApiVersions.GetApiVersion(options.FakeRestApiVersions().FakeResourceVersion.ResourceType.ToString());
+            Assert.True(result.Equals(FakeResourceApiVersions.V2019_12_01));
```

Comment:
This appears to be 4 different tests which we can separate out, and lets add inside / outside / on boundary cases

Reply:
Really just one I think, testing change from default, back to default, to a custom string, back to a another defined API versions.

Reply:
maybe its two?
1. Validate setting via extension affects TryGetApiVersion
2. Validate setting via SetApiVersion affects TryGetApiVersion?